module/VuFind/src/VuFind/Favorites/FavoritesService.php

```diff
@@ -135,13 +132,13 @@ public function destroyList(
             throw new ListPermissionException('list_access_denied');
         }
 
-        // Remove user_resource and resource_tags rows:
-        $userResource = $this->getDbTable('UserResource');
-        $userResource->destroyLinks(null, $list->getUser()?->getId(), $list->getId());
+        // Remove user_resource and resource_tags rows for favorites tags:
+        $listUser = $list->getUser();
+        $this->resourceTagsService->destroyResourceTagsLinksForUser(null, $listUser, $list);
```

Comment:
Previously the destroyLinks() method was automatically calling the tag destruction method. It seemed safer to separate these things for clarity; since we now encapsulate the functionality within FavoritesService, this seemed reasonable.